packages/nodejs/src/transmitter.ts

```diff
@@ -0,0 +1,96 @@
+import fs from "fs"
+import path from "path"
+import https from "https"
+import http from "http"
+
+import { Configuration } from "./config"
+import { URL, URLSearchParams } from "url"
+import { HashMap } from "@appsignal/types"
+
+export class Transmitter {
+  #config: Configuration
+  #url: string
+  #data: string
+
+  constructor(url: string, data = "", active = true) {
+    this.#config = new Configuration({ active })
+    this.#url = url
+    this.#data = data
+  }
+
+  public async transmit() {
+    return new Promise<HashMap<any>>((resolve, reject) => {
+      const config_data = this.#config.data
+      const params = new URLSearchParams({
+        api_key: config_data["pushApiKey"] || "",
+        name: config_data["name"] || "",
+        environment: config_data["environment"] || "",
+        hostname: config_data["hostname"] || ""
+      })
+
+      const url = new URL(`${this.#url}?${params.toString()}`)
+      const opts = this.requestOptions(url)
+      const requestModule = url.protocol == "http:" ? http : https
+
+      const request = requestModule.request(opts, (stream: any) => {
+        const responseStatus = stream.statusCode
+        stream.setEncoding("utf8")
+        let responseBody = ""
+
+        stream
+          .on("data", (chunk: any) => {
+            responseBody += chunk
+          })
+          .on("end", () => {
+            let parsedResponse
+            try {
+              parsedResponse = JSON.parse(responseBody)
+            } catch {
+              parsedResponse = {}
+            }
+
+            if (responseStatus == 200) {
+              resolve({ status: responseStatus, body: parsedResponse })
+            } else {
+              reject({ status: responseStatus, body: parsedResponse })
+            }
+          })
+      })
+
+      request.write(this.#data)
+      request.end()
+    })
+  }
+
+  private requestOptions(requestUrl: URL): HashMap<any> {
+    const configData = this.#config.data
+
+    return {
+      method: "POST",
+      protocol: requestUrl.protocol,
+      host: requestUrl.hostname,
+      port: requestUrl.port,
+      path: requestUrl.toString(),
+      headers: {
+        "Content-Type": "application/json",
+        "Content-Length": this.#data.length
+      },
+      ca: this.certificateFile()
+    }
+  }
+
+  private certificateFile(): string {
+    const configData = this.#config.data
+    const caFilePathFromConfig = configData["caFilePath"] || ""
+    let caFilePath
```

Comment:
I updated it to `let statusCode = null` on the latest changes. This must be here because the variable value is assigned from an async function using `await`. Our TS config is strict, so `strictNullChecks` are activated by default. We have to initialize these variables as "something" (`null` in this case) to make the compiler happy.

Reply:
But we don't use the variable anywhere right? So we can remove this entire line.